misc/dbt-materialize/tests/adapter/test_ci.py

```diff
@@ -0,0 +1,309 @@
+            fetch="one",
+        )
+        assert (
+            result is not None
+        ), "Cluster with manual schedule was not created successfully"
+
+        # Cleanup
+        run_dbt(
+            [
+                "run-operation",
+                "drop_cluster",
+                "--args",
+                '{"cluster_name": "test_manual_schedule_dbt_deploy"}',
+            ]
+        )
+
+    def test_create_cluster_with_on_refresh_schedule(self, project):
+        # Test creating a cluster with on-refresh schedule type
+        run_dbt(
+            [
+                "run-operation",
+                "create_cluster",
+                "--args",
+                '{"cluster_name": "test_on_refresh_schedule", "size": "1", "schedule_type": "on-refresh", "refresh_rehydration_time_estimate": "10m", "ignore_existing_objects": true, "force_deploy_suffix": true}',
+            ]
+        )
+
+        # Verify cluster creation
+        result = project.run_sql(
+            "SELECT name FROM mz_clusters WHERE name = 'test_on_refresh_schedule_dbt_deploy'",
+            fetch="one",
+        )
+        assert (
+            result is not None
+        ), "Cluster with on-refresh schedule was not created successfully"
+
+        # Cleanup
+        run_dbt(
+            [
+                "run-operation",
+                "drop_cluster",
+                "--args",
+                '{"cluster_name": "test_on_refresh_schedule_dbt_deploy"}',
+            ]
+        )
+
+    def test_create_cluster_without_force_deploy_suffix(self, project):
+        # Test creating a cluster without force deploy suffix
+        run_dbt(
+            [
+                "run-operation",
+                "create_cluster",
+                "--args",
+                '{"cluster_name": "test_cluster", "size": "1", "replication_factor": 1, "ignore_existing_objects": true, "force_deploy_suffix": false}',
+            ]
+        )
+
+        # Verify cluster creation
+        result = project.run_sql(
+            "SELECT name FROM mz_clusters WHERE name = 'test_cluster'",
+            fetch="one",
+        )
+        assert (
+            result is not None
+        ), "Cluster without force deploy suffix was not created successfully"
+
+        # Cleanup
+        run_dbt(
+            [
+                "run-operation",
+                "drop_cluster",
+                "--args",
+                '{"cluster_name": "test_cluster"}',
+            ]
+        )
+
+    def test_create_cluster_with_replication_and_schedule(self, project):
+        # Test creating a cluster with both replication factor and schedule type
+        run_dbt(
+            [
+                "run-operation",
+                "create_cluster",
+                "--args",
+                '{"cluster_name": "test_cluster", "size": "1", "replication_factor": 2, "schedule_type": "manual", "refresh_rehydration_time_estimate": "10m", "ignore_existing_objects": true, "force_deploy_suffix": true}',
```

Comment:
I'm confused by this test. 🤔 It seems like it's testing that you shouldn't be able to use `refresh_rehydration_time_estimate` with `manual`, not `schedule_type` and `replication_factor` (which would then be okay to use with `manual`, just not `on-schedule`).